internal/collections/noop.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2023 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package collections
+
+import (
+	"github.com/corazawaf/coraza/v3/collection"
+	"github.com/corazawaf/coraza/v3/types"
+)
+
+type Noop struct{}
+
+var _ collection.Noop = &Noop{}
+
+// NewNoop creates a new Noop.
+func NewNoop() *Noop {
```

Comment:
Probably we can just declare a variable and reuse it instead of creating new pointers